lib/collectionspace/converter/ohc/collectionobject.rb

```diff
@@ -0,0 +1,139 @@
+module CollectionSpace
+  module Converter
+    module OHC
+      class OHCCollectionObject < AnthroCollectionObject
+        ::OHCCollectionObject = CollectionSpace::Converter::OHC::OHCCollectionObject
+        def convert
+          run(wrapper: 'document') do |xml|
+            xml.send(
+              "ns2:collectionobjects_common",
+              "xmlns:ns2" => "http://collectionspace.org/services/collectionobject",
+              "xmlns:xsi" => "http://www.w3.org/2001/XMLSchema-instance"
+            ) do
+              xml.parent.namespace = nil
+              CoreCollectionObject.map(xml, clean_overridden_attributes)
+              OHCCollectionObject.map_common_overrides(xml, attributes)
+            end
+
+            xml.send(
+              "ns2:collectionobjects_anthro",
+              "xmlns:ns2" => "http://collectionspace.org/services/collectionobject/domain/anthro",
+              "xmlns:xsi" => "http://www.w3.org/2001/XMLSchema-instance"
+            ) do
+              xml.parent.namespace = nil
+              AnthroCollectionObject.map(xml, clean_overridden_attributes)
+            end
+
+            xml.send(
+              "ns2:collectionobjects_ohc",
+              "xmlns:ns2" => "http://collectionspace.org/services/collectionobject/domain/ohc",
+              "xmlns:xsi" => "http://www.w3.org/2001/XMLSchema-instance"
+            ) do
+              xml.parent.namespace = nil
+              OHCCollectionObject.map(xml, attributes)
+            end
+          end
+        end #def convert
+
+        # This is used instead of the redefined_fields method because using that method did not
+        #  work here, where ohc calls both anthro and core, and both ohc an anthro override different
+        #  core field conversion behavior. Using redefined_fields in ohc was resulting in only
+        #  anthro's overrides being applied. 
+        def clean_overridden_attributes
+          overridden = [
+            'assocpeople',
+            'assocpeopletype',
+            'assocpeoplenote',
+            'objectnametype',
+            'objectnamesystem',
+            'objectname',
+            'objectnamecurrency',
+            'objectnamenote',
+            'objectnamelevel',
+            'objectnamelanguage'
+          ]
+          safe = attributes.clone
```

Comment:
@mark-cooper -- So, if I did not use '.clone' here, `safe.delete(fieldname)` also affected the main attributes hash. Attributes I needed to send through the special OHC overrides were no longer there. 

I knew I'd ascertained the necessity of `.clone` in some situations... Do you know off the top of your head what the difference it here, as opposed to the copy of the string in the DateParser?

Reply:
@kspurgin I believe it's just implementation differences for data structures that is generally in line with how other languages work. For standard data types (ints, strings etc.) there's generally copy type behavior and for data structures (array, hash etc.) there's generally reference type behavior (even though for Ruby everything is an object behind the scenes).